ResearchKit/Common/ORKFormItemCell.m

```diff
@@ -1062,3 +1068,171 @@ - (BOOL)textField:(UITextField *)textField shouldChangeCharactersInRange:(NSRang
+    } else {
+        _placeholder = formItem.placeholder;
+    }
+    
+    [self applyAnswerFormat];
+}
+
+- (void)answerDidChange {
+    id answer = self.answer;
+    if (answer == ORKNullAnswerValue()) {
+        answer = nil;
+    }
+    _selectionView.answer = (NSValue *)answer;
+    [_selectionView setTextColor:[UIColor blackColor]];
+}
+
+- (void)setEditingHighlight:(BOOL)editingHighlight {
+    _editingHighlight = editingHighlight;
+    [_selectionView setTextColor:( _editingHighlight ? [self tintColor] : [UIColor blackColor])];
+}
+
+- (void)selectionViewDidBeginEditing:(ORKLocationSelectionView *)view {
+    self.editingHighlight = YES;
+    [_selectionView showMapViewAnimated:YES];
+    [self.delegate formItemCellDidBecomeFirstResponder:self];
+}
+
+- (void)selectionViewDidEndEditing:(ORKLocationSelectionView *)view {
+    self.editingHighlight = NO;
+    [_selectionView hideMapViewAnimated:YES];
+    [self.delegate formItemCellDidResignFirstResponder:self];
+}
+
+- (void)selectionViewSelectionDidChange:(ORKLocationSelectionView *)view {
+    [self inputValueDidChange];
+}
+
+- (void)selectionViewNeedsResize:(ORKLocationSelectionView *)view {
+    UITableView *tableView = [self parentTableView];
+    
+    [tableView beginUpdates];
+    [tableView endUpdates];
+    
+    CGRect convertedVisibleRect = [tableView convertRect:view.bounds fromView:view];
+    [tableView scrollRectToVisible:convertedVisibleRect animated:YES];
+}
+
+- (void)selectionViewError:(NSError *)error {
+    [self showErrorAlertWithTitle:ORKLocalizedString(@"LOCATION_ERROR_TITLE", @"") message:error.localizedDescription];
+}
+
+- (void)inputValueDidChange {
+    [self ork_setAnswer:((_selectionView.answer != nil) ? _selectionView.answer : ORKNullAnswerValue())];
+    [super inputValueDidChange];
+}
+
+- (BOOL)becomeFirstResponder {
+    return [_selectionView becomeFirstResponder];
+}
+
+- (BOOL)resignFirstResponder {
+    BOOL resign = [super resignFirstResponder];
+    resign = [_selectionView resignFirstResponder] || resign;
+    return resign;
+}
+
+- (BOOL)isAnswerValid {
+    id answer = _selectionView.answer;
+    
+    if (answer == ORKNullAnswerValue()) {
+        return YES;
+    }
+    
+    ORKAnswerFormat *answerFormat = [self.formItem impliedAnswerFormat];
+    ORKLocationAnswerFormat *locationFormat = (ORKLocationAnswerFormat *)answerFormat;
+    
+    NSNumberFormatter *decimalFormatter = [[NSNumberFormatter alloc] init];
+    decimalFormatter.numberStyle = NSNumberFormatterDecimalStyle;
+    
+    CLLocationCoordinate2D location = ((NSValue *)answer).MKCoordinateValue;
+    
+    NSNumber *latitude = [NSNumber numberWithDouble:location.latitude];
+    NSNumber *longitude = [NSNumber numberWithDouble:location.longitude];
+    
+    NSString *string = [NSString stringWithFormat:@"%@, %@", [decimalFormatter stringFromNumber:latitude], [decimalFormatter stringFromNumber:longitude]];
+    
```

Comment:
Move this code that generates the string into a method, since it is duplicated down below inside `shouldContinue`.